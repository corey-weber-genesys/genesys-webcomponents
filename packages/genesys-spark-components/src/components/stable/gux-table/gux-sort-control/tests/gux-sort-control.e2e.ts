import { E2EPage, newE2EPage } from '@stencil/core/testing';

import { a11yCheck } from '../../../../../test/e2eTestUtils';

import markup from './gux-sort-control.markup';

async function newNonrandomE2EPage(
  {
    html
  }: {
    html: string;
  },
  lang: string = 'en'
): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(`<div lang=${lang}>${html}</div>`);
  await page.waitForChanges();
  await page.addScriptTag({
    path: '../../node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-sort-control', () => {
  describe('#render', () => {
    [
      markup.notSorted,
      markup.sortedAscending,
      markup.sortedDescending,
      markup.sortedDescendingUnsortedAllowed
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-sort-control');
        await a11yCheck(page);

        expect(element).toHaveAttribute('hydrated');
        expect(element.shadowRoot.innerHTML).toMatchSnapshot();
      });
    });
  });

  describe('#events', () => {
    describe('guxsortchanged', () => {
      [
        markup.notSorted,
        markup.sortedAscending,
        markup.sortedDescending,
        markup.sortedDescendingUnsortedAllowed
      ].forEach(({ description, html, onclickSortDirection }) => {
        it(description, async () => {
          const page = await newNonrandomE2EPage({ html });
          await page.waitForChanges();

          const guxsortchangedSpy = await page.spyOnEvent('guxsortchanged');

          const th = await page.find('th[data-column-name="c1"]');
          await th.click();

          expect(guxsortchangedSpy).toHaveFirstReceivedEventDetail({
            columnName: 'c1',
            sortDirection: onclickSortDirection
          });
        });
      });
    });
  });
});
