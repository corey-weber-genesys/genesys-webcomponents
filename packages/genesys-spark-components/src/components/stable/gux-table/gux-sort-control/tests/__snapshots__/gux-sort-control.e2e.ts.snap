// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-sort-control #render Not sorted 1`] = `"<div class=\\"gux-container\\"><button class=\\"gux-sort-button\\" type=\\"button\\"><span class=\\"gux-sr-only\\">Column Header 1 . The table is not currently sorted by this column. Click the column header to sort the table by this column in ascending order</span><gux-icon class=\\"gux-sort-icon\\" icon-name=\\"arrow-solid-down\\" hydrated=\\"\\"></gux-icon></button><div class=\\"gux-resize-spacer\\"></div></div>"`;

exports[`gux-sort-control #render Sorted ascending 1`] = `"<div class=\\"gux-container\\"><button class=\\"gux-sort-button gux-active\\" type=\\"button\\"><span class=\\"gux-sr-only\\">Column Header 1 . The table is currently sorted by this column in ascending order. Click the column header to sort the table by this column in descending order</span><gux-icon class=\\"gux-sort-icon\\" icon-name=\\"arrow-solid-down\\" hydrated=\\"\\"></gux-icon></button><div class=\\"gux-resize-spacer\\"></div></div>"`;

exports[`gux-sort-control #render Sorted descending 1`] = `"<div class=\\"gux-container\\"><button class=\\"gux-sort-button gux-active\\" type=\\"button\\"><span class=\\"gux-sr-only\\">Column Header 1 . The table is currently sorted by this column in descending order. Click the column header to sort the table by this column in ascending order</span><gux-icon class=\\"gux-sort-icon\\" icon-name=\\"arrow-solid-up\\" hydrated=\\"\\"></gux-icon></button><div class=\\"gux-resize-spacer\\"></div></div>"`;

exports[`gux-sort-control #render Sorted descending, unsorted allowed 1`] = `"<div class=\\"gux-container\\"><button class=\\"gux-sort-button gux-active\\" type=\\"button\\"><span class=\\"gux-sr-only\\">Column Header 1 . The table is currently sorted by this column in descending order. Click the column header to not sort the table by this column</span><gux-icon class=\\"gux-sort-icon\\" icon-name=\\"arrow-solid-up\\" hydrated=\\"\\"></gux-icon></button><div class=\\"gux-resize-spacer\\"></div></div>"`;
